core/src/main/scala/org/apache/spark/sql/SnappyStrategies.scala

```diff
@@ -301,7 +317,7 @@ private[sql] object JoinStrategy {
  *
  * Adapted from Spark's Aggregation strategy.
  */
-object SnappyAggregation extends Strategy {
+private[sql] object SnappyAggregationStrategy extends Strategy {
```

Comment:
The problem was for the session object, which is not available in SnappyAggregationStrategy. SnappyStrategies is constructed with a DefaultPlanner , hence could get hold of the session object easily. Not so with SnappyAggregationStrategy. Did not want to do a lot of refactoring to get the flag.